File: 백준/Silver/1929. 소수 구하기/소수 구하기.py
M, N = map(int, input().split())

ptr = 0
prime = [None] * 300000

prime[ptr] = 2
ptr += 1
prime[ptr] = 3
ptr += 1

for n in range(5, N + 1, 2):
  i = 1
  while prime[i] * prime[i] <= n:  
    if n % prime[i] == 0:
      break
    i += 1
  else:
    prime[ptr] = n
    ptr += 1       
    
for i in range(ptr):
  if M <= prime[i] <= N:
    print(prime[i])

File: 백준/Silver/1929. 소수 구하기/소수 구하기.java
import java.io.*;

class Main {
    public static void main(String[] args) throws IOException {
        try (BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
             BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out))) {
            String[] MN = br.readLine().split(" ");
            int m = Integer.parseInt(MN[0]);
            int n = Integer.parseInt(MN[1]);
            int[] prime = new int[n + 1];
            StringBuilder sb = new StringBuilder();
            for (int i = 0; i <= n; i++) {
                prime[i] = i;
            }
            for (int i = 2; i * i <= n; i++) {
                if (prime[i] == 0)
                    continue;
                for (int j = i + i; j <= n; j = j + i) {
                    prime[j] = 0;
                }
            }
            for (int i = m; i <= n; i++) {
                if (i == 1)
                    continue;
                if (prime[i] != 0)
                    sb.append(prime[i] + "\n");
            }
            bw.write(sb.toString());
        }
    }
}